waspc/src/Wasp/Lib.hs

```diff
@@ -36,52 +37,67 @@ compile ::
   CompileOptions ->
   IO ([CompileWarning], [CompileError])
 compile waspDir outDir options = do
-  appSpecOrCompileErrors <- analyzeWaspProject waspDir options
+  (compileWarnings, appSpecOrCompileErrors) <- analyzeWaspProject waspDir options
   case appSpecOrCompileErrors of
-    Left compileErrors -> return ([], compileErrors)
+    Left compileErrors -> return (compileWarnings, toList compileErrors)
     Right appSpec ->
       case ASV.validateAppSpec appSpec of
         [] -> do
           (generatorWarnings, generatorErrors) <- Generator.writeWebAppCode appSpec outDir (sendMessage options)
-          return (map show $ generatorWarningsFilter options generatorWarnings, map show generatorErrors)
+          let filteredGeneratorWarnings = map show $ generatorWarningsFilter options generatorWarnings
+          return (filteredGeneratorWarnings ++ compileWarnings, map show generatorErrors)
         validationErrors -> do
-          return ([], map show validationErrors)
+          return (compileWarnings, map show validationErrors)
```

Comment:
Is this any better? I tried extracting analyzerWarnings out. But not sure if improvement! But I do think maybe naming those errors/warnings from analyzer as analyzerWarnings/errors makes sense.

```hs
compile waspDir outDir options = do
  (analyzerWarnings, appSpecOrAnalyzerErrors) <- analyzeWaspProject waspDir options
  compilerWarningsAndErrors <- case appSpecOrAnalyzerErrors of
    Left analyzerErrors -> return ([], toList analyzerErrors)
    Right appSpec ->
      case ASV.validateAppSpec appSpec of
        [] -> do
          (generatorWarnings, generatorErrors) <- Generator.writeWebAppCode appSpec outDir (sendMessage options)
          let filteredGeneratorWarnings = map show $ generatorWarningsFilter options generatorWarnings
          return (filteredGeneratorWarnings, map show generatorErrors)
        validationErrors -> do
          return ([], map show validationErrors)
    return $ (analyzerWarnings, []) <> compilerWarningsAndErrors
```

Reply:
I updated both of those functions to reduce that warning repetition. Thanks again for the suggestion! 👍🏻